src/Chunk.ts

```diff
@@ -280,14 +282,20 @@ export default class Chunk {
 		options: OutputOptions,
 		existingNames: Record<string, any>
 	): string {
-		const sanitizedId = sanitizeFileName(this.orderedModules[0].id);
+		const id = this.orderedModules[0].id;
+		const sanitizedId = sanitizeFileName(id);
 
 		let path: string;
-		if (isAbsolute(this.orderedModules[0].id)) {
+		if (isAbsolute(id)) {
+			const extension = extname(id);
+			console.log({ COMMON_JS_EXTENSIONS, extension });
 			const name = renderNamePattern(
-				options.entryFileNames || '[name].js',
+				options.entryFileNames ||
+					(COMMON_JS_EXTENSIONS.includes(extension) ? '[name].js' : '[name].[ext].js'),
```

Comment:
Not sure why the name is `COMMON_JS...` as I do not see the connection. How about `JAVASCRIPT_EXTENSION` or `IGNORED_EXTENSIONS`?  Also, I would rather have the default pattern to be `[name][extname].js`, the reason being that otherwise, files without extension would have two subsequent `.` in their name.

Reply:
`JAVASCRIPT_EXTENSIONS` would work, just seemed a little bit inaccurate as it also includes TS extensions - i'm ok with renaming it though to whatever you see consider fitting here

> Also, I would rather have the default pattern to be [name][extname].js

I've wanted to avoid output files being named like `Buttonscss.js`, like - having some separator seems visually nice

> files without extension would have two subsequent . in their name.

Gonna add some special handling for it + tests.

Reply:
> I've wanted to avoid output files being named like Buttonscss.js, like - having some separator seems visually nice

Definitely, but the point of `[extname]` vs `[ext]` should be that `[extname]` contains a leading `.` if there is an extension and is an empty string otherwise. So it basically solves exactly this issue. So

| filename | ext | extname | `[name][extname].js` | `[name].[ext].js`
| ---           | --- | --- | --- | ---
| `"test.css"`      | `"css"` | `".css"` | `"test.css.js"` | `"test.css.js"`
| `"test"`     | `""` | `""` | `"test.js"` | `"test..css"` ⚠️

Reply:
> i'm ok with renaming it though to whatever you see consider fitting here

How about `NON_ASSET_EXTENSIONS`?

Reply:
> Definitely, but the point of [extname] vs [ext] should be that [extname] contains a leading . if there is an extension and is an empty string otherwise. 

Cool, didn't notice I could use it like this.

> How about NON_ASSET_EXTENSIONS?

Sure, sounds good!